src/main.rs

```diff
@@ -119,22 +143,58 @@ fn work(args: Args) -> Result<(), Error> {
         }
     }
 
-    let license = if must_copy_license(&config.package.license) {
-        p("LICENSE file will be installed manually.".bold().yellow());
-        Some(license_file()?)
-    } else {
-        None
-    };
-
     if args.dryrun.not() {
-        release_build(args.musl)?;
-        tarball(args.musl, &cargo_target, &output, license.as_ref(), &config)?;
-        let sha256: String = sha256sum(&config.package, &output)?;
+        let source = args.source.unwrap_or_default();
+        let (sha256, license) = match (source, args.no_bin) {
+            (Source::CratesIo, true) => {
+                let crate_file = crates::CrateFile::download_new(&config)?;
+                let sha256 = crate_file.get_sha256sum()?;
+                let license = crate_file.get_license()?;
+                (sha256, license)
+            }
+            (Source::CratesIo, false) => {
+                let crate_file = crates::CrateFile::download_new(&config)?;
+                let built_crate_file = crate_file.build(args.musl)?;
+                let license = built_crate_file.tarball(&cargo_target)?;
+                let sha256 = sha256sum(&config.package, &output)?;
+                (sha256, license)
+            }
+            (Source::Project, true) => {
+                source_tarball(&cargo_target, &output, &config)?;
+                let license = if must_copy_license(&config.package.license) {
+                    p("LICENSE file will be installed manually.".bold().yellow());
+                    Some(license_file(None)?)
+                } else {
+                    None
+                };
```

Comment:
This same logic is repeated in several places (and the `p` message), and I suspect this can be consolidated.

Reply:
actioned